src/Nerdbank.Streams/MultiplexingStream.Channel.cs

```diff
@@ -381,7 +381,7 @@ public void Dispose()
                         // We can't know whether there is or will be more bytes written to the user's PipeWriter,
                         // but we need to terminate our reader for their writer as part of reclaiming resources.
                         // Cancel the pending or next read operation so the reader loop will immediately notice and shutdown.
-                        this.mxStreamIOReader.CancelPendingRead();
+                        this.mxStreamIOReader?.CancelPendingRead();
```

Comment:
is it ok that you can get dual cancellation here?

Reply:
My understanding is we are not expecting this type to be called concurrently, but it is for some reason. For now we are mitigating the specific exception that was causing it to invalidate state further up the state.

Reply:
It's a good question. I'm not certain, but I'm pretty sure this one method can handle concurrent execution since its purpose is to be called to cancel another operation.